internal/mssql/sqlclient.go

```diff
@@ -34,6 +36,36 @@ func NewClient(host string, port int64, database string, username string, passwo
 	return c
 }
 
+// NewAzureADClient creates a new SQL client authenticated via Azure AD.
+func NewAzureADClient(host string, port int64, database string) (SqlClient, error) {
+	if port <= 0 {
+		port = 1433
+	}
+
+	// Set up Managed Identity credential using system-assigned identity
+	cred, err := azidentity.NewManagedIdentityCredential(nil)
```

Comment:
Would it be better to use `DefaultAzureCredential` to support other authentication methods as well or this is intended just for system-managed identities?

Reply:
Another note is that you may want to configure the the az.ClientOptions for the credential in the cases of sovereign clouds (US Government, Germany, China). Something like this:

```go
azureCloud := getEnv("AZURE_CLOUD", "AzureUSGovernmentCloud")
var cc cloud.Configuration
if azureCloud == "AzureUSGovernmentCloud" {
cc = cloud.AzureGovernment
}
if azureCloud == "AzurePublicCloud" {
cc = cloud.AzurePublic
}
opts := azcore.ClientOptions{Cloud: cc}
cred, err := azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{ClientOptions: opts})
```